x-pack/plugins/apm/kibana.jsonc

```diff
@@ -29,7 +29,8 @@
       "lens",
       "maps",
       "uiActions",
-      "observabilityAIAssistant"
+      "observabilityAIAssistant",
+      "observabilityLogExplorer"
```

Comment:
So are we envisioning that the observability log explorer will never want to use anything from APM, metrics, profiling, etc? This might be another plugin to plugin dependency anti-pattern we're now introducing here. Is this mainly for accessing locators?

Reply:
OK, I see locator sharing is happening a lot more, which is great, but I think as an observability org we should decide on the best way for doing that, whether it's using IDs like you mention, putting all locators in observability_shared or another "safe" dependency, or otherwise. 

In the meantime, what worries me here is that we've added this dependency which allows other features to take advantage of it, potentially making it much more difficult to unwind. Is it simple to remove the dependency and use the ID for now?

Reply:
I agree with @jasonrhodes about accessing the locators by ID, although the recommended way to consume a locator is from the start contract, each of those sets the basis for a circular dependency if the plugins might share some logic.
I followed the same practice for the "Add data" button that navigates from the observability log explorer plugin to the onboarding one, I'd recommend using the ID since the beginning to keep the dependencies limited.

Reply:
Yes, I hadn't seen the ID pattern but if you need to consume the ID value from the other plugin, this may not solve the dependency problem. Ultimately, I expect we need to store all locators in a central place such as observability_shared, to make cross-plugin use possible. 

- [ ] @tonyghiani are you saying if we switch to an ID access here, we wouldn't be able to remove the APM -> obs log explorer dependency?

Reply:
@jasonrhodes
> I expect we need to store all locators in a central place such as observability_shared, to make cross-plugin use possible.

I believe what you mean here is what already happens with the `share` plugin, which allows to register locators into a registry.
By "using the locator ID", I mean consuming the locator from this registry instead of the start contract of the observability-log-explorer start contract, which would remove the dependency and use the `share` plugin instead (which for apm is already used)